source/AI.cpp

```diff
@@ -3359,32 +3359,118 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 		// Board the nearest disabled ship, focusing on hostiles before allies. Holding
 		// `Shift` results in boarding only player-owned escorts in need of assistance.
 		shared_ptr<const Ship> target = ship.GetTargetShip();
-		if(!target || !CanBoard(ship, *target) || (shift && !target->IsYours()))
+		if(!target || (target && activeCommands.Has(Command::WAIT)) || !CanBoard(ship, *target)
+				|| (shift && !target->IsYours()))
 		{
 			if(shift)
 				ship.SetTargetShip(shared_ptr<Ship>());
 
-			double closest = numeric_limits<double>::infinity();
 			bool foundEnemy = false;
-			bool foundAnything = false;
-			for(const shared_ptr<Ship> &other : ships)
-				if(CanBoard(ship, *other))
+			const auto targetPriority = Preferences::BoardingSetting();
+			const bool costPriority = (targetPriority == "value");
+			const bool distancePriority = (targetPriority == "proximity");
+
+			auto strategy = [&]() noexcept -> function<double(Ship &)>
+			{
+				Point current = ship.Position();
+				if(costPriority)
 				{
-					if(shift && !other->IsYours())
-						continue;
+					return [this, &ship](Ship &other) noexcept -> double
+					{
+						// Use the exact cost if the ship was scanned, otherwise use an estimation.
+						return this->Has(ship, other.shared_from_this(), ShipEvent::SCAN_OUTFITS) ?
+							other.Cost() : (other.ChassisCost() * 2.);
+					};
+				}
+				else if(distancePriority)
+				{
+					return [current](Ship &other) noexcept -> double
+					{
+						return current.DistanceSquared(other.Position());
+					};
+				} // Default to mixed strategy.
```

Comment:
This probably shouldn't be the default.
The default for boarding friendly or escort ships is nearest first. The default for selecting ships is also nearest first.
Unless you've changes those, too? Which doesn't seem like a great thing to do.

Reply:
But thats the nearest option?

Reply:
It still is the case.

It defaults to this if the word is not recognized as distance or cost

Reply:
Updated the comment

Reply:
Still don't think it should use 'mixed' as a fallback.

Reply:
Fair, I mostly wanted to do this because the name for mixed could always change, but since its parsed earlier into this enum class I'm fine with it not being default here.